### PR TITLE
Attempt to explain why Mocktail should be ordered

### PR DESCRIPTION
**Pull request by calebhearth**
This is mostly a rephrasing of the existing section. My goal is to make
it easier for a developer to glance at this section and say: "yeah,
Mocktail seems like the right choice for my project"

This can be considered a starting point - I'm happy to incorporate more /
different reasons or to take a different approach.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,20 +50,26 @@ assert_equal "🎉", result
...
-  [custom matchers](#custom-matchers)
-* [Argument captors](#mocktailcaptor) for assertions of very complex arguments
+## Why order?
+
+Besides a lack of hangover, Mocktail offers several advantages over other
+mocking libraries:
+
+* **Fewer hoops to jump through**: [`Mocktail.of_next(type)`] avoids the need
+  for dependency injection by returning a Mocktail of the type the next time
+  `Type.new` is called. You can inject a fake into use production code in one
```
Comment (pingortle) [inline README.md:60]:
```suggestion
  `Type.new` is called. You can inject a fake into production code in one
```

Typo or I can't read? Just taking a guess at the intended phrasing here. 😄

File: README.md [hunk 2]
```diff
@@ -50,20 +50,26 @@ assert_equal "🎉", result
...
-* Dynamic stubbings based on the arguments passed to the actual call
-* Built-in [argument matchers](#mocktailmatchers) and a simple API for adding
-  [custom matchers](#custom-matchers)
-* [Argument captors](#mocktailcaptor) for assertions of very complex arguments
+## Why order?
+
+Besides a lack of hangover, Mocktail offers several advantages over other
+mocking libraries:
+
+* **Fewer hoops to jump through**: [`Mocktail.of_next(type)`] avoids the need
```
Comment (pingortle) [inline README.md:58]: FWIW I'm glad you're putting this one front and center. It was the first big feature/benefit combo that smacked me in the face when perusing the docs. 💯

**Review by pingortle — COMMENTED** [review]
👋 Pardon the drive-by review.

I came here thinking this PR was gonna somehow run my tests in order. 😆  Glad to see it was a different meaning of "order".

FWIW, looks good to this internet rando so far. 😄 👍

File: README.md (hunk 2 above)
Comment (calebhearth) [reply README.md:58]: Yeah if nothing else changed, I think putting this here is the thing I'd still like to see happen.

Comment (calebhearth) [reply README.md:58]: Yeah if nothing else changed, I think putting this here is the thing I'd still like to see happen.

**Comment by calebhearth** [comment]
@searls

**Pull request merged**